### PR TITLE
Add link to v2 migration document and Auth nuget to Contents in README for v2 branch

### PR DESCRIPTION
**Pull request by brycewang-microsoft**

### CODE REVIEW

File: readme.md [hunk 1]
```diff
@@ -8,6 +16,7 @@ This repository contains the following:
 - **Microsoft Azure IoT Hub service SDK for C#** to manage your IoT Hub service instance from a back-end .NET application.
 - **Microsoft Azure Provisioning device SDK for C#** to provision devices to Azure IoT Hub with .NET.
 - **Microsoft Azure Provisioning service SDK for C#** to manage your Provisioning service instance from a back-end .NET application.
+- **Microsoft Azure IoT Devices Authentication Providers for C#** to provide Authentication codes for Azure IoT device SDKs.
```
Comment (timtay-microsoft) [inline readme.md:19]:
```suggestion
- **Microsoft Azure IoT Devices Authentication Providers for C#** to provide authentication class for provisioning devices and connecting to IoT hub.
```

**Review by timtay-microsoft — APPROVED** [review]

File: readme.md [hunk 2]
```diff
@@ -1,5 +1,13 @@
 # Microsoft Azure IoT SDK for .NET
 
+## 2.0.0 clients release notice
+
+A [2.0.0 preview release](https://github.com/Azure/azure-iot-sdk-csharp/releases/tag/2022-10-14-v2preview) has been published for each package in this library.
```
Comment (drwill-ms) [inline readme.md:5]: Do not encourage users to migrate to a preview release!

Instead, we want them to check it out and give us feedback.

**Review by drwill-ms — APPROVED** [review]

**Pull request merged**